tests/e2e/common/mod.rs

```diff
@@ -24,30 +24,37 @@ static CONFIG: Lazy<Config> = Lazy::new(Config::parse_env);
 /// Generates a client using the active config.
 ///
 /// This is a function rather than a `Lazy` because every executor (IE, [`#[tokio::test]`](tokio::test)) needs its own client.
-fn client() -> Client {
+fn client() -> (bool, Client) {
     let config = &*CONFIG;
+    let mut is_local = false;
 
     let client = match Client::for_name(&config.network_name) {
-        Ok(client) => client,
+        Ok(client) => {
+            if &config.network_name == "localhost" {
+                is_local = true
+            }
+
+            client
+        }
         Err(e) => {
             // to ensure we don't spam the logs with `Error creating client: ...`,
             // we just let an arbitrary thread win and log the "error".
             static LOGS_ONCE: AtomicBool = AtomicBool::new(false);
 
             // note: Relaxed is probably fine, AcqRel is *definitely* fine.
             if !LOGS_ONCE.swap(true, std::sync::atomic::Ordering::AcqRel) {
-                log::error!("Error creating client: {e}; creating one using `testnet`");
+                log::error!("Error creating client: {e}; creating one using `localhost`");
```

Comment:
testnet should still be the default, it's almost always going to be available, but local node might not